CHANGELOG.md

```diff
@@ -1,6 +1,127 @@
+     required int32 id = 2;
+     optional string email = 3;
+   }
+   ```
+   
+   the generated Kotlin code will look like the following:
+   
+   ```kotlin
+   data class Person(
+     @field:WireField(tag = 1, adapter = "com.squareup.wire.ProtoAdapter#STRING") val name: String,
+     @field:WireField(tag = 2, adapter = "com.squareup.wire.ProtoAdapter#INT32") val id: Int,
+     @field:WireField(tag = 3, adapter = "com.squareup.wire.ProtoAdapter#STRING") val email: String? =
+         null,
+     val unknownFields: ByteString = ByteString.EMPTY
+   ) : Message<Person, Person.Builder>(ADAPTER, unknownFields) {
+     companion object {
+       @JvmField
+       val ADAPTER: ProtoAdapter<Person> = ... // code omitted for brevity
+   ```
+   
+   The `copy()` method of a data class replaces most usages of the builder. If your code relies on 
+   the `Builder` you can enable full `Builder` generation by passing the `--java_interop` parameter 
+   to the compiler.
+ 
+ * New: gRPC support
+ 
+   In addition to generating Kotlin code from proto messages, Wire can now generate code for gRPC
+   endpoints. Here's an example schema:
+   
+   ```proto
+   service RouteGuide {
+     // A simple RPC.
+     //
+     // Obtains the feature at a given position.
+     //
+     // A feature with an empty name is returned if there's no feature at the given
+     // position.
+     rpc GetFeature(Point) returns (Feature) {}
+   }    
+   ```
+   
+   The generated code will look like the following (message protos, referenced by the schema, are
+   omitted):
+   
+   ```kotlin
+   interface RouteGuide : Service {
+     @WireRpc(
+         path = "/routeguide.RouteGuide/GetFeature",
+         requestAdapter = "routeguide.Point#ADAPTER",
+         responseAdapter = "routeguide.Feature#ADAPTER"
+     )
+     suspend fun GetFeature(request: Point): Feature
+   }
+   ```
+   
+   All four gRPC modes are supported: the generated code uses suspendable functions to implement
+   non-blocking asynchronous execution. In streaming modes, `ReceiveChannel` and `SendChannel` are
+   used to listen to asynchronous data in a non-blocking fashion.
+   
+   This feature works out-of-the-box in Wire 3 compiler as long as the input file contains a gRPC
+   schema.
+ 
+ * New: Gradle plugin
+ 
+   Here's an example of Gradle configuration:
+   
+   ```groovy
+   apply plugin: 'com.squareup.wire'
+   
+   wire {
+     // Keeps only 'Dinosaur#name' as the root of the object graph
+     roots 'squareup.dinosaurs.Dinosaur#name'
+   
+     // Keeps all fields, except 'name', in 'Dinosaur'
+     prunes 'squareup.dinosaurs.Dinosaur#name'
+   
+     // Both roots and prunes in an external file
+     rules 'rules.txt'
+   
+     kotlin {
+       javaInterop true
+       out "${buildDir}/generated/custom"
+     }
+   }
+   ```
+   
+   The `wire` extension introduces the concept of compilation targets, such as `kotlin` and `java`,
+   where each target has its own configuration properties.
```

Comment:
Maybe add a note that both a java and android target can be supplied to ease migration efforts?